### PR TITLE
Add semantic diffs between hardforks to the docs.

### PR DESCRIPTION
**Pull request by SamWilsn**
### What was wrong?

No easy way to see changes between hard forks.

### How was it fixed?

Using a modified version of `rstdiff` from the docutils package.

Won't really show much until a new hard fork is added, but I've done some limited testing with fake forks locally.

Couple known issues:

 - Links inside the diffed documents point to `.pickle` files instead of `.html`, which don't exist.
 - Very noisy diff since anywhere the module name appears, there's a change reported.
 - No quick navigation to the diffs.

Here's what an example change might look like so far: https://samwilsn.github.io/eth1.0-specs/diffs/frontier_banana/rlp/index.html

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

@voith would you also mind taking a look?

### CODE REVIEW

File: src/ethereum_spec_tools/diff.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    old_path = old.name.replace(".", os.sep)
+    old_path = os.path.join(input_path, old_path)
+
+    new_path = new.name.replace(".", os.sep)
+    new_path = os.path.join(input_path, new_path)
+
+    diff_path = old.name.split(".")[-1] + "_" + new.name.split(".")[-1]
+    diff_path = os.path.join(output_path, diff_path)
+
+    old_pickles = set(find_pickles(old_path, old))
```
Comment (lightclient) [inline src/ethereum_spec_tools/diff.py:55]: Just curious, why return an iterator and create a set vs. simply returning a set? More flexibility?

File: src/ethereum_spec_tools/diff.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    new_pickles = set(find_pickles(new_path, new))
+
+    pickles = old_pickles | new_pickles
+
+    for count, pickle_path in enumerate(pickles, start=1):
+        old_pickle_path = os.path.join(old_path, pickle_path)
+        new_pickle_path = os.path.join(new_path, pickle_path)
+        diff_pickle_path = os.path.join(diff_path, pickle_path + "64")
```
Comment (lightclient) [inline src/ethereum_spec_tools/diff.py:63]: Why `64`?

**Review by lightclient — APPROVED** [review]
Looks good to me.

File: whitelist.txt [hunk 3]
```diff
@@ -274,3 +274,12 @@ noop
 0x0
 0x1
 0x2
+
+rstdiff
+docutils
+dirname
+Text2
```
[the remark this answers is not in the document]
Comment (SamWilsn) [reply whitelist.txt:281]: Yeppers. Probably worth doing some work upstream to improve it a bit.

File: src/ethereum_spec_tools/diff.py (hunk 1 above)
Comment (SamWilsn) [reply src/ethereum_spec_tools/diff.py:55]: Didn't really think much about it. Seems a bit clunky to write:

```python
x = set()
for y in z:
    x.append(x)
return x
```

File: src/ethereum_spec_tools/diff.py (hunk 2 above)
Comment (SamWilsn) [reply src/ethereum_spec_tools/diff.py:63]: The build process has two stages, `stage0` and `stage1`.

`stage0` writes to binary pickle files, and those files are opened by `rstdiff`, which is fine handling binary files.

Sphinx, on the other hand, only lets you define parsers in extensions and not readers. Parsers can only handle strings (not `bytes`) so the input has to be valid UTF-8.

The output from the `rstdiff` is written to `.pickle64` files (base64 encoded pickle) to satisfy that requirement.

Comment (lightclient) [reply src/ethereum_spec_tools/diff.py:63]: ahh okay gotcha!

File: src/ethereum_spec_tools/diff.py (hunk 1 above)
Comment (lightclient) [reply src/ethereum_spec_tools/diff.py:55]: okay that makes sense, you're version is a lot more terse

**Pull request merged**